text-generation-inference/server/text_generation_server/cli.py

```diff
@@ -19,6 +19,7 @@ def serve(
     logger_level: str = "INFO",
     json_output: bool = False,
     otlp_service_name: str = "text-generation-inference.server",
+    max_input_tokens: Optional[int] = None,
```

Comment:
> I think you do not need this, if you set the env var MAX_INPUT_LENGTH

@tengomucho I tried using `--max-input-length` with `text-generation-launcher`  but received error as below.  Could you please elaborate on how we can use MAX_INPUT_LENGTH as env var. Also TGI documentation mentions max_input_length as legacy version of max_input_tokens, will this be a concern?

```# text-generation-launcher --port 8000 --max-concurrent-requests 1 --max-input-length 512 --max-total-tokens 1024 --max-batch-prefill-tokens 512 --max-batch-total-tokens 1024
2024-08-13T16:04:40.833385Z  INFO text_generation_launcher: Args {
    model_id: "meta-llama/Meta-Llama-3.1-8B",
    ...
    ...
}
...

2024-08-13T16:04:43.826483Z  INFO download: text_generation_launcher: Successfully downloaded weights for meta-llama/Meta-Llama-3.1-8B
2024-08-13T16:04:43.826735Z  INFO shard-manager: text_generation_launcher: Starting shard rank=0
2024-08-13T16:04:43.927753Z ERROR shard-manager: text_generation_launcher: Shard complete standard error output:

Usage: text-generation-server serve [OPTIONS] MODEL_ID
Try 'text-generation-server serve --help' for help.

Error: No such option: --max-input-tokens rank=0
2024-08-13T16:04:44.026753Z ERROR text_generation_launcher: Shard 0 failed to start
2024-08-13T16:04:44.026765Z  INFO text_generation_launcher: Shutting down shards
Error: ShardCannotStart```

Reply:
you are right, the right parameter to use is "MAX_INPUT_TOKENS", I mixed things up. My point is: what do you want to achieve by adding this to the cli in the text generation server? For now IIRC this value is used in the launcher to define the maximum number of input tokens that can be passed from the router to the server. The server for now does not use that. It is ok to add it to the cli, but to be effective you will also need to add it to the `serve` function, and do something with it, otherwise it will not have any effect.